core/primitives/transaction_validity.hpp

```diff
@@ -152,14 +152,23 @@ namespace kagome::primitives {
   }
 
   /// An unknown transaction validity.
-  enum class UnknownTransaction : uint8_t {
-    /// Could not lookup some information that is required to validate the
-    /// transaction.
-    CannotLookup = 1,
-    /// No validator found for the given unsigned transaction.
-    NoUnsignedValidator,
-    /// Any other custom unknown validity that is not covered by this enum.
-    Custom
+  struct UnknownTransaction {
+    enum Kind : uint8_t {
+      /// Could not lookup some information that is required to validate the
+      /// transaction.
+      CannotLookup = 1,
+      /// No validator found for the given unsigned transaction.
+      NoUnsignedValidator,
+      /// Any other custom unknown validity that is not covered by this enum.
+      Custom
+    };
+
+    bool operator==(Kind kind) const {
+      return this->kind == kind;
```

Comment:
Remove or compare `custom_value` (because rust `#[derive(Eq)]`)

Reply:
I'm not sure I quite understand what you mean. This function is needed to simplify checking if an UnknownTransaction object belongs to a certain category.